lib/rules/no-broken-super-chain.js

```diff
@@ -3,93 +3,107 @@
  */
 
 const MESSAGES = {
-  noSuper: '\'this._super(...arguments)\' must be called in init()',
-  noThisBeforeSuper: 'Must call \'this._super(...arguments)\' before accessing `this`',
-  tooManySupers: 'Only call `this._super(...arguments)` once per init()'
+  noSuper: '`this._super(...arguments)` must be called in',
+  tooManySupers: 'Only call `this._super(...arguments)` once per lifecycle hook',
+  argsNotPassedToSuper: '...arguments need to be passed to this._super() call'
 };
 
-// TODO: Make this configurable
-const EMBER_MODULES_WITH_SUPER_CHAIN = {
-  Component: true,
-  Mixin: true,
-  Route: true,
-  Controller: true,
-  View: true
-};
+const LIFECYCLE_HOOKS = ['init', 'willDestroy', 'willDestroyElement', 'destroy'];
 
-/**
- * Determines if this is an init method in an extension of Ember[EMBER_MODULES_WITH_SUPER_CHAIN.*]
- * @param {Node} node
- */
-function isInit(node) {
-  if (node.type === 'FunctionExpression' && node.parent && node.parent.key && node.parent.key.name === 'init') {
+function isExtend(node) {
+  return node && node.callee && node.callee.property && node.callee.property.name === 'extend';
+}
+
+function isSuperCall(lineWithinFn) {
+  if (lineWithinFn.type !== 'MemberExpression') {
+    return false;
+  }
+
+  let isSuperCall = false;
+
+  if (lineWithinFn.object.type === 'ThisExpression') {
+    isSuperCall = lineWithinFn.property.type === 'Identifier' && lineWithinFn.property.name === '_super';
+  } else if (lineWithinFn.object.type === 'MemberExpression') {
+    isSuperCall = lineWithinFn.object.property.name === '_super';
+  }
+
+  return isSuperCall;
+}
 
-    if (node.parent.parent
-      && node.parent.parent.parent
-      && node.parent.parent.parent.callee
-      && node.parent.parent.parent.callee.object
-      && node.parent.parent.parent.callee.object.object
-      && node.parent.parent.parent.callee.object.object.name === 'Ember') {
-      return (node.parent.parent.parent.callee.object.property
-        && EMBER_MODULES_WITH_SUPER_CHAIN[node.parent.parent.parent.callee.object.property.name]);
-    }
+function wereArgumentsPassedToSuper(expression) {
+  const callee = expression.callee;
+
+  if (!expression || callee.type !== 'MemberExpression') {
+    return;
   }
 
-  return false;
+  if (callee.property.name === '_super') {
+    const firstArgumentToSuper = expression.arguments[0];
+    return firstArgumentToSuper && firstArgumentToSuper.type === 'SpreadElement' && firstArgumentToSuper.argument.name === 'arguments';
+  } else if (callee.property.name === 'apply') {
+    const args = expression.arguments;
+    return args.length >= 2 && args[0].type === 'ThisExpression' && args[1].name === 'arguments';
```

Comment:
When would `args` be `> 2`?  I think we want `args.length === 2`...

Reply:
+1

Reply:
Yeah, not sure what I had in mind here